.travis.yml

```diff
@@ -43,131 +43,132 @@ env:
+        # Get last remaining tags, if any.
+        - git fetch --tags origin
+
+        # Test if code was formatted with 'go fmt'
+        # Command will format code and return modified files
+        # fail if any have been modified.
+        - if [ -n "$(go fmt)" ]; then echo 'Code is not formatted with "go fmt"'; false; fi
+
+        # Perform static code analysys
+        - go vet `go list ./... | grep -v vendor`
+
+        # Fail builds when the cyclomatic complexity reaches 15 or more
+        - gocyclo -over 15 `find . -iname '*.go' | grep -v 'vendor' | grep -v '_test.go'`
+
+        # Verify that the Swagger docs are valid
+        - swagger-cli validate docs/*.yml
+
+        # Verify that the Swagger docs follow the autodeployment requirements
+        - if test "$(ls -A docs)"; then python2.7 verify_docs.py `find docs -name "*.yml"`; fi
+    - stage: Tests
+      name: "unit"
+      install: skip
+      script:
+        - go list ./... | grep -v vendor | xargs -n1 -I {} -P 4 go test -v -covermode=atomic -coverprofile=../../../{}/coverage.txt {} || exit $? ;
+
+      after_success:
+        - bash <(curl -s https://codecov.io/bash) -F unittests ;
+    - stage: Tests
+      name: "acceptance"
+      install: skip
+      script:
+        - git clone -b master https://github.com/mendersoftware/integration.git;
+        - mv integration/extra/travis-testing/* tests/ ;
+
+        - CGO_ENABLED=0 go test -c -o useradm-test -coverpkg $(go list ./... | grep -v vendor | grep -v mocks | grep -v test | tr  '\n' ,);
+
+        - sudo docker build -f Dockerfile.acceptance-testing -t mendersoftware/useradm:prtest .;
+
+        - go build ;
+        - ./tests/build-acceptance ./tests ./docs/internal_api.yml ./docs/management_api.yml useradm;
+
+        - TESTS_DIR=$PWD/tests ./tests/run-test-environment "acceptance" $PWD/integration ./tests/docker-compose-acceptance.yml && TESTS_DIR=$PWD/tests ./tests/run-test-environment "acceptance" $PWD/integration ./tests/docker-compose-acceptance-multitenant.yml;
+      after_success:
+        - bash <(curl -s https://codecov.io/bash) -F acceptance -f ./tests/coverage-acceptance.txt;
+    - stage: Tests
+      name: "backend-integration"
+      install:
+        - sudo apt-get -qq update
+        - pip3 install -U --user PyYAML
+      script:
+        - git clone -b master https://github.com/mendersoftware/integration.git;
+
+        - CGO_ENABLED=0 go build;
+        - sudo docker build -t mendersoftware/useradm:pr .;
+
+        - ./integration/extra/release_tool.py --set-version-of mender-useradm --version pr;
+        - ./integration/backend-tests/run;
+    - stage: Build and publish
+      name: "dockerhub"
+      install: skip
+      script:
+        - CGO_ENABLED=0 go build -ldflags "-X main.Commit=`echo $TRAVIS_COMMIT` -X main.Tag=`echo $TRAVIS_TAG` -X main.Branch=`echo $TRAVIS_BRANCH` -X main.BuildNumber=`echo $TRAVIS_BUILD_NUMBER`" ;
+      deploy:
```

Comment:
I think deploy doe not support custom commands, before_deploy + dymmy deploy that always triggers were used in other places.

Reply:
how to define a dummy `deploy` step? we used to actually have a valid s3 provider deploy - now it lives in a different stage.

for now I just added before_deploy, I'm not sure if it will be called at all though (doc says before_deploy is ran once per provider - no providers, no deploys).

Reply:
ok, apparently there's a `script` deploy provider, and it should be enough to give it `/bin/true`